### ISSUE TITLE
07 -- Camille Mauban

### ISSUE THREAD

**Issue by cmauban** [TIY-Durham/2015-FALL-FEE#155]
- **Yak Shaving**
  - Create a new _WIP Issue_ named `07 -- YOUR NAME` with:
    - [x] link to your PR in `TIY-Assignments` from `07--the-last-cascade` into `master`
- **Reading JavaScript: Events**
  - [x] How does an `Event` "bubble"? Can I make it stop?
  - [x] What does the `target` property of an `Event` refer to?
  - [x] Besides `click`, what _other_ event types can I trigger with my mouse?
  - [x] What types of events can be triggered on a `<form>` element?
  - [x] `click` me, baby, one more time!
  - [x] `scroll` down the avenue
  - [x] `change` is a-comin'
  - [x] `submit` to mah authah-ri-tah!
  - [x] this is a complete `load`...
  - [x] well, just `unload` about it.
  - [ ] **BEAST MODE** dat keyboard, tho
- **Building the Blueprints**
  - `NestedAccordions` can...
    - [x] First accordion opens when closed... It's a start!
    - [x] First accordion closes when open... It's a finish?
    - [x] First accordion opens _and_ closes! It's a wrap! :shipit:
    - [x] Top-level accordions can open and close... :astonished:
    - [ ] **BEAST MODE** Accordion ALL THE THINGS!
  - `ViewModeSwitch` can...
    - **Always start with the button...**
      - [x] TRACER BULLETS: Houston, we are clear for clicking...
      - [x] Clicking switches the `"active"` button...
    - **Welcome to the Grid, User...**
      - [x] TRACER BULLETS: What is changing? Can you change that?
      - [x] Now make a _Function_ that does that...
    - **Wiring up the clicks...**
      - [x] Click to switch the grid layout to a list...
      - [x] Click to switch the list layout to a grid...
  - `MultiColumnForm` can...
    - **Who are you listening to?**
      - [x] TRACER BULLETS: I can see the `target`...
      - [x] It's not the `<button>`, BTW.
      - [x] Ready for the main `Event`!
    - **Where is the data?**
      - [x] Capture the form submission with JavaScript
      - [x] Then output the form data to the console
    - [x] Post the output to a comment in your _WIP Issue_

**Comment by cmauban** [comment]
07 Blueprints
https://github.com/cmauban/TIY-Assignments/pull/11

**Comment by cmauban** [comment]
JavaScript Cheatsheet
https://github.com/cmauban/TIY-Assignments/pull/6

**Comment by cmauban** [comment]
```
scripts.js:30 camille
scripts.js:30 mauban
scripts.js:30 camille.mauban@yas.com
scripts.js:30 Italy
scripts.js:30 im a cool kid
scripts.js:30 9199066514
scripts.js:30 tiy
scripts.js:30 Web Designer
scripts.js:30 kitty kat
scripts.js:30 iphone
scripts.js:30 Ninja silence
scripts.js:30 Oj
scripts.js:30 invisible
scripts.js:30 love
scripts.js:30 Yassss, did this work!!??
scripts.js:30 Send your data
```

**Comment by cmauban** [comment]
This weekend started off a little rough for me. I spent all day and night on saturday trying to figure out `ViewModeSwitch`. My problem was that I was making it wayyy too complicated. Once I got NodeList down from saturday, it made doing the `MultiColumnForm` much easier to figure out. Yes! - finally getting it. I just wish I had enough time to do one more of one of these assignments so I can do it completely on my own so it can stick.
